src/Switch/Switch.js

```diff
@@ -31,6 +31,10 @@ const Switch = React.forwardRef(({
 
     let [isChecked, setIsChecked] = useState(!!checked);
 
+    useEffect(() => {
+        if (checked) setIsChecked(checked);
+    }, [checked]);
+
```

Comment:
@rjstyles Can you add a test for this? Thanks!

Reply:
I think we'd want to remove the `if(checked)` so it could set the state to checked=false
```jsx
    useEffect(() => {
        setIsChecked(checked);
    }, [checked]);
```

Reply:
> @rjstyles Can you add a test for this? Thanks!

Sure, will do that

Reply:
Yeah @skvale, Thanks for the review, we can do something like below to also handle  if `checked` is `null` or `undefined`

```
useEffect(() => {
    setIsChecked(!!checked);
}, [checked]);
```

@jbadan I will add the tests for this.